src/lib/tile-processing/tile3d/builders/Tile3DMultipolygon.ts

```diff
@@ -199,34 +201,54 @@ export default class Tile3DMultipolygon {
 		return this.cachedPoleOfInaccessibility;
 	}
 
-	public populateWithPoints(resolution: number, tileSize: number): Vec2[] {
-		const tiles = this.getCoveredTiles(resolution, tileSize);
+	public populateWithPoints(resolution: number, pointsPerBox: number, seed: number): Vec2[] {
+		const tiles = this.getCoveredTiles(resolution, Config.TileSize);
 		const points: Vec2[] = [];
+		const rnd = new SeededRandom(seed);
 
 		for (const tile of tiles) {
 			const [x, y] = tile.split(' ').map(v => +v);
-			const point = new Vec2(
-				(x + 0.75 - Math.random() * 0.5) / resolution * tileSize,
-				(y + 0.75 - Math.random() * 0.5) / resolution * tileSize,
-			);
 
-			let isInMultipolygon = true;
+			for (let i = 0; i < pointsPerBox; ++i) {
+				const offset = 0.5;
+				const factor = 1.0;
+				const point = new Vec2( // 0.75 - Math.random() * 0.5
+					(x + offset - rnd.generate() * factor) / resolution * Config.TileSize,
+					(y + offset - rnd.generate() * factor) / resolution * Config.TileSize,
+				);
+
+				let isInMultipolygon = true;
 
-			for (const ring of this.rings) {
-				if (ring.type === Tile3DRingType.Outer) {
-					if (!ring.isContainsPoints(point)) {
+				// check if to close to the other points
+				for (const other of points) {
```

Comment:
Do you think we can optimize this? Comparing distance to every other point naively like this is not an option, it really hurts performance. I did a rough benchmark and for some polygons it can take 100-200 ms to execute this method, while the old implementation takes <5 ms.

Reply:
Oh, I made a mistake! Only the trees of one plant-tile should get checked.
But as wie now have only a view plants per tile, we can drop that check at all.

Reply:
We can't drop it, trees must have some space between them. With pure random we will get a lot of trees placed really close to each other.
Also, it's still possible for trees from nearby tiles to be very close to trees from the current tile if they are near the tile border.

Reply:
Did you try it? OSM2World runs without it and looks fine. Well, my opinion only.

Reply:
I don't like that the distribution is not really uniform, there's a lot of empty spots and there are spots with multiple trees clipping through each other.

Maybe we can try using Halton sequence here instead of random?

Reply:
You young people have great ideas. There is even code:
https://en.wikipedia.org/wiki/Halton_sequence

Reply:
Halton worked at once for Forests. So we don’t need a distance check. Should run fast.
We even may not use the “plant-tiles” any more, but it works with it too.
I needed some time to understand the 5 bushes groups are not caused by my code but are your instance group. This reduces the number of instances but makes it more difficult to avoid gaps in the shrubbery.
Even the forest may have “natural” gaps. I prefer this to get nice random trees. You could change the parameters. If we don’t need different Tree- and BushTiles.
As the code has changed, I will make a new pull request and close this one
OK?

Reply:
It's better to use tiles because they give an approximation of the polygon so that we generate fewer nodes outside of the polygon (although it's still possible for a node to end up outside of the polygon, hence we need to calculate `isInMultipolygon`).

There's no need to make a new PR, you can just push more changes into this one.